bindings/python/cntk/layers/sequence.py

```diff
@@ -666,3 +666,65 @@ def unfold_from(initial_state, dynamic_axes_like):
         return output
 
     return _inject_name(unfold_from, name)
+
+
+def QRNN(filter_shape=2, hidden_dim=None, activation=C.tanh, return_full_state=False):
+    """
+    Quasi-Recurrent Neural Networks
+
+    This is the CNTK implementation of [Salesforce Research](https://einstein.ai/)'s
+    [Quasi-Recurrent Neural Networks](https://arxiv.org/abs/1611.01576) paper.
+
+    From the authors:
+        The QRNN provides similar accuracy to the LSTM but can be between
+        2 and 17 times faster than the highly optimized NVIDIA cuDNN LSTM
+        implementation depending on the use case.
+
+        If you use this code or our results in your research, please cite:
+
+        @article{bradbury2016quasi,
+          title={{Quasi-Recurrent Neural Networks}},
+          author={Bradbury, James and Merity, Stephen and Xiong, Caiming and Socher, Richard},
+          journal={International Conference on Learning Representations (ICLR 2017)},
+          year={2017}
+        }
+
+    Args:
+        filter_shape: integer, the size of convolution across the temporal
+        dynamic axis. size 2 reduces down to LSTM implementation.
+        hidden_dim: integer, size of hidden dim of h, c and o
+        activation: cell activation function
+
+    Returns:
+        :class:`~cntk.ops.functions.Function`: OR
+        tuple of :class:`~cntk.ops.functions.Function`:
+
+    """
+
+    def FoPool(c, fz):
+        f = C.slice(fz, 0, 0, hidden_dim)
+        z = C.slice(fz, 0, hidden_dim, 2 * hidden_dim)
+        return f * c + (1 - f) * z
+
+    def model(input_tensor):
+        gate_values = Convolution(filter_shape=filter_shape, num_filters=3 * hidden_dim, pad=True,
+                                  reduction_rank=1, sequential=True)(input_tensor)
```

Comment:
Recently we have updated and added a new interface C.layer.SequentialConvolution([link](https://github.com/Microsoft/CNTK/blob/3806ec6e0ebb3f47ac56cd937a5b9f3dcf3108d8/bindings/python/cntk/layers/layers.py#L278)) for more complete support of sequential convolution. It'd be great if you could have a look and switch to the new api. It should work the same by switching the called method name to SequentialConvolution and drop the parameter "sequential=True".

Reply:
Updated to the new API. Thanks for pointing that out!